<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="Tests API" tests="87" failures="34" errors="0" time="16.826">
  <testsuite name="tests\unit\models\Entreprise.test.js" errors="0" failures="0" skipped="1" timestamp="2025-10-26T15:41:09" time="3.913" tests="24">
    <testcase classname="Entreprise Model › Schema Validation" name="should create a valid entreprise with minimal data" time="0.187">
    </testcase>
    <testcase classname="Entreprise Model › Schema Validation" name="should require nomEntreprise" time="0.022">
    </testcase>
    <testcase classname="Entreprise Model › Schema Validation" name="should require region" time="0.016">
    </testcase>
    <testcase classname="Entreprise Model › Schema Validation" name="should validate region enum" time="0.013">
    </testcase>
    <testcase classname="Entreprise Model › Schema Validation" name="should require ville" time="0.013">
    </testcase>
    <testcase classname="Entreprise Model › Schema Validation" name="should require dateCreation" time="0.009">
    </testcase>
    <testcase classname="Entreprise Model › Schema Validation" name="should require secteurActivite" time="0.014">
    </testcase>
    <testcase classname="Entreprise Model › Schema Validation" name="should validate secteurActivite enum" time="0.012">
    </testcase>
    <testcase classname="Entreprise Model › Schema Validation" name="should require sousSecteur" time="0.013">
    </testcase>
    <testcase classname="Entreprise Model › Schema Validation" name="should require formeJuridique" time="0.008">
    </testcase>
    <testcase classname="Entreprise Model › Schema Validation" name="should require numeroContribuable" time="0.013">
    </testcase>
    <testcase classname="Entreprise Model › Schema Validation" name="should validate numeroContribuable format" time="0.012">
    </testcase>
    <testcase classname="Entreprise Model › Schema Validation" name="should enforce unique numeroContribuable" time="0.061">
    </testcase>
    <testcase classname="Entreprise Model › Schema Validation" name="should require contact email" time="0.007">
    </testcase>
    <testcase classname="Entreprise Model › Schema Validation" name="should validate contact email format" time="0.015">
    </testcase>
    <testcase classname="Entreprise Model › Performance Economique Validation" name="should validate chiffreAffaires montant is not negative" time="0.01">
    </testcase>
    <testcase classname="Entreprise Model › Performance Economique Validation" name="should validate coutsProduction montant is not negative" time="0.011">
    </testcase>
    <testcase classname="Entreprise Model › Investissement Emploi Validation" name="should require effectifsEmployes" time="0.013">
    </testcase>
    <testcase classname="Entreprise Model › Innovation Digitalisation Validation" name="should validate integrationInnovation range" time="0.013">
    </testcase>
    <testcase classname="Entreprise Model › Conventions Validation" name="should validate atteinteCiblesInvestissement range" time="0.015">
    </testcase>
    <testcase classname="Entreprise Model › Default Values" name="should set default statut to En attente" time="0.017">
    </testcase>
    <testcase classname="Entreprise Model › Default Values" name="should set default conformite to Non vérifié" time="0.015">
    </testcase>
    <testcase classname="Entreprise Model › Default Values" name="should set default informationsCompletes to false" time="0.02">
    </testcase>
    <testcase classname="Entreprise Model › Virtual Properties" name="should populate conventionsActives virtual" time="0">
      <skipped/>
    </testcase>
  </testsuite>
  <testsuite name="tests\unit\models\User.test.js" errors="0" failures="0" skipped="0" timestamp="2025-10-26T15:41:09" time="6.742" tests="20">
    <testcase classname="User Model › Schema Validation" name="should create a valid user" time="0.416">
    </testcase>
    <testcase classname="User Model › Schema Validation" name="should require nom field" time="0.019">
    </testcase>
    <testcase classname="User Model › Schema Validation" name="should require prenom field" time="0.014">
    </testcase>
    <testcase classname="User Model › Schema Validation" name="should require email field" time="0.012">
    </testcase>
    <testcase classname="User Model › Schema Validation" name="should require valid email format" time="0.009">
    </testcase>
    <testcase classname="User Model › Schema Validation" name="should require motDePasse field" time="0.01">
    </testcase>
    <testcase classname="User Model › Schema Validation" name="should enforce minimum password length" time="0.013">
    </testcase>
    <testcase classname="User Model › Schema Validation" name="should enforce unique email" time="0.366">
    </testcase>
    <testcase classname="User Model › Schema Validation" name="should validate role enum" time="0.016">
    </testcase>
    <testcase classname="User Model › Schema Validation" name="should validate typeCompte enum" time="0.005">
    </testcase>
    <testcase classname="User Model › Password Hashing" name="should hash password before saving" time="0.119">
    </testcase>
    <testcase classname="User Model › Password Hashing" name="should not hash password if not modified" time="0.13">
    </testcase>
    <testcase classname="User Model › Instance Methods" name="should compare password correctly" time="0.437">
    </testcase>
    <testcase classname="User Model › Instance Methods" name="should return public profile without password" time="0.159">
    </testcase>
    <testcase classname="User Model › Instance Methods" name="should check if user is admin" time="0.14">
    </testcase>
    <testcase classname="User Model › Virtual Properties" name="should return full name" time="0.133">
    </testcase>
    <testcase classname="User Model › Static Methods" name="should find user by email (case insensitive)" time="0.163">
    </testcase>
    <testcase classname="User Model › Default Values" name="should set default role to user" time="0.166">
    </testcase>
    <testcase classname="User Model › Default Values" name="should set default status to active" time="0.261">
    </testcase>
    <testcase classname="User Model › Default Values" name="should set default entrepriseIncomplete to false" time="0.189">
    </testcase>
  </testsuite>
  <testsuite name="tests\integration\auth\auth.register.test.js" errors="0" failures="11" skipped="0" timestamp="2025-10-26T15:41:09" time="8.343" tests="12">
    <testcase classname="Auth Registration Integration Tests › POST /api/auth/register" name="should register a new entreprise user successfully" time="0.831">
      <failure>Error: expected 201 &quot;Created&quot;, got 500 &quot;Internal Server Error&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.register.test.js:27:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Auth Registration Integration Tests › POST /api/auth/register" name="should register a new admin user successfully" time="0.646">
      <failure>Error: expect(received).toBeUndefined()

Received: null
    at Object.toBeUndefined (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.register.test.js:68:52)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)</failure>
    </testcase>
    <testcase classname="Auth Registration Integration Tests › POST /api/auth/register" name="should reject registration with existing email" time="0.296">
    </testcase>
    <testcase classname="Auth Registration Integration Tests › POST /api/auth/register" name="should reject registration with invalid email format" time="0.048">
      <failure>Error: expected 400 &quot;Bad Request&quot;, got 500 &quot;Internal Server Error&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.register.test.js:116:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Auth Registration Integration Tests › POST /api/auth/register" name="should reject registration with short password" time="0.043">
      <failure>Error: expected 400 &quot;Bad Request&quot;, got 500 &quot;Internal Server Error&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.register.test.js:134:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Auth Registration Integration Tests › POST /api/auth/register" name="should reject registration with missing required fields" time="0.061">
      <failure>Error: expected 400 &quot;Bad Request&quot;, got 500 &quot;Internal Server Error&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.register.test.js:150:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Auth Registration Integration Tests › POST /api/auth/register" name="should create entreprise with default values for entreprise user" time="0.045">
      <failure>Error: expected 201 &quot;Created&quot;, got 500 &quot;Internal Server Error&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.register.test.js:167:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Auth Registration Integration Tests › POST /api/auth/register" name="should hash password before saving" time="0.042">
      <failure>Error: expected 201 &quot;Created&quot;, got 500 &quot;Internal Server Error&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.register.test.js:194:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Auth Registration Integration Tests › POST /api/auth/register" name="should set default role to user for entreprise account" time="0.055">
      <failure>Error: expected 201 &quot;Created&quot;, got 500 &quot;Internal Server Error&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.register.test.js:213:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Auth Registration Integration Tests › POST /api/auth/register" name="should set entrepriseIncomplete to true for entreprise user" time="0.089">
      <failure>Error: expected 201 &quot;Created&quot;, got 500 &quot;Internal Server Error&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.register.test.js:231:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Auth Registration Integration Tests › POST /api/auth/register" name="should generate unique numeroContribuable for entreprise" time="0.067">
      <failure>Error: expected 201 &quot;Created&quot;, got 500 &quot;Internal Server Error&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.register.test.js:249:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Auth Registration Integration Tests › POST /api/auth/register" name="should handle registration with special characters in names" time="0.063">
      <failure>Error: expected 201 &quot;Created&quot;, got 500 &quot;Internal Server Error&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.register.test.js:269:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
  </testsuite>
  <testsuite name="tests\integration\auth\auth.login.test.js" errors="0" failures="6" skipped="0" timestamp="2025-10-26T15:41:09" time="11.321" tests="12">
    <testcase classname="Auth Login Integration Tests › POST /api/auth/login" name="should login with valid credentials" time="1.602">
      <failure>Error: expect(received).toContain(expected) // indexOf

Matcher error: received value must not be null nor undefined

Received has value: undefined
    at Object.toContain (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.login.test.js:35:37)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)</failure>
    </testcase>
    <testcase classname="Auth Login Integration Tests › POST /api/auth/login" name="should login admin user successfully" time="0.503">
      <failure>Error: expect(received).toBeDefined()

Received: undefined
    at Object.toBeDefined (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.login.test.js:65:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)</failure>
    </testcase>
    <testcase classname="Auth Login Integration Tests › POST /api/auth/login" name="should login entreprise user with entreprise data" time="0.046">
      <failure>ValidationError: Entreprise validation failed: investissementEmploi.effectifsEmployes: Le nombre d&apos;employés est requis
    at model.Document.invalidate (D:\api-gestion-main\New folder\server\node_modules\mongoose\lib\document.js:3358:32)
    at D:\api-gestion-main\New folder\server\node_modules\mongoose\lib\document.js:3119:17
    at D:\api-gestion-main\New folder\server\node_modules\mongoose\lib\schemaType.js:1416:9
    at processTicksAndRejections (node:internal/process/task_queues:85:11)</failure>
    </testcase>
    <testcase classname="Auth Login Integration Tests › POST /api/auth/login" name="should reject login with invalid email" time="0.08">
    </testcase>
    <testcase classname="Auth Login Integration Tests › POST /api/auth/login" name="should reject login with invalid password" time="0.491">
    </testcase>
    <testcase classname="Auth Login Integration Tests › POST /api/auth/login" name="should reject login with missing email" time="0.059">
    </testcase>
    <testcase classname="Auth Login Integration Tests › POST /api/auth/login" name="should reject login with missing password" time="0.054">
    </testcase>
    <testcase classname="Auth Login Integration Tests › POST /api/auth/login" name="should handle case insensitive email" time="0.467">
    </testcase>
    <testcase classname="Auth Login Integration Tests › POST /api/auth/login" name="should return JWT token with correct payload" time="0.5">
      <failure>Error: expect(received).toBeDefined()

Received: undefined
    at Object.toBeDefined (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.login.test.js:222:21)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)</failure>
    </testcase>
    <testcase classname="Auth Login Integration Tests › POST /api/auth/login" name="should not return password in response" time="0.51">
    </testcase>
    <testcase classname="Auth Login Integration Tests › POST /api/auth/login" name="should handle inactive user" time="0.588">
      <failure>Error: expected 401 &quot;Unauthorized&quot;, got 200 &quot;OK&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.login.test.js:274:10)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Auth Login Integration Tests › POST /api/auth/login" name="should handle pending user" time="0.521">
      <failure>Error: expected 401 &quot;Unauthorized&quot;, got 200 &quot;OK&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\auth\auth.login.test.js:299:10)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
  </testsuite>
  <testsuite name="tests\integration\entreprises\entreprise.crud.test.js" errors="0" failures="17" skipped="0" timestamp="2025-10-26T15:41:09" time="15.87" tests="19">
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › GET /api/entreprises" name="devrait obtenir toutes les entreprises pour l&apos;administrateur" time="1.501">
      <failure>Error: expect(received).toBe(expected) // Object.is equality

Expected: true
Received: undefined
    at Object.toBe (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:72:37)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › GET /api/entreprises" name="devrait filtrer les entreprises par statut" time="0.543">
      <failure>Error: expect(received).toBe(expected) // Object.is equality

Expected: true
Received: undefined
    at Object.toBe (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:105:37)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › GET /api/entreprises" name="devrait filtrer les entreprises par région" time="0.519">
      <failure>Error: expect(received).toBe(expected) // Object.is equality

Expected: true
Received: undefined
    at Object.toBe (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:115:37)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › GET /api/entreprises" name="devrait exiger une authentification" time="0.47">
      <failure>Error: expected 401 &quot;Unauthorized&quot;, got 200 &quot;OK&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:122:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › GET /api/entreprises" name="devrait exiger le rôle administrateur" time="0.544">
      <failure>Error: expected 403 &quot;Forbidden&quot;, got 200 &quot;OK&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:134:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › GET /api/entreprises/:id" name="devrait obtenir une entreprise par ID pour l&apos;administrateur" time="0.583">
      <failure>Error: expected 200 &quot;OK&quot;, got 401 &quot;Unauthorized&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:146:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › GET /api/entreprises/:id" name="devrait obtenir sa propre entreprise pour l&apos;utilisateur entreprise" time="0.523">
      <failure>Error: expected 200 &quot;OK&quot;, got 401 &quot;Unauthorized&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:159:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › GET /api/entreprises/:id" name="ne devrait pas permettre à un utilisateur entreprise d&apos;accéder à d&apos;autres entreprises" time="0.495">
      <failure>Error: expected 403 &quot;Forbidden&quot;, got 401 &quot;Unauthorized&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:191:10)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › GET /api/entreprises/:id" name="devrait retourner 404 pour une entreprise inexistante" time="0.407">
      <failure>Error: expected 404 &quot;Not Found&quot;, got 401 &quot;Unauthorized&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:203:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › PUT /api/entreprises/:id" name="devrait mettre à jour une entreprise pour l&apos;administrateur" time="0.467">
      <failure>Error: expected 200 &quot;OK&quot;, got 401 &quot;Unauthorized&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:225:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › PUT /api/entreprises/:id" name="devrait mettre à jour sa propre entreprise pour l&apos;utilisateur entreprise" time="0.783">
      <failure>Error: expected 200 &quot;OK&quot;, got 401 &quot;Unauthorized&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:246:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › PUT /api/entreprises/:id" name="ne devrait pas permettre à un utilisateur entreprise de mettre à jour d&apos;autres entreprises" time="0.315">
      <failure>Error: expected 403 &quot;Forbidden&quot;, got 401 &quot;Unauthorized&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:284:10)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › PUT /api/entreprises/:id" name="devrait valider les champs obligatoires" time="0.352">
      <failure>Error: expected 400 &quot;Bad Request&quot;, got 401 &quot;Unauthorized&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:301:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › POST /api/entreprises" name="devrait créer une nouvelle entreprise pour l&apos;administrateur" time="0.517">
      <failure>TypeError: Cannot read properties of undefined (reading &apos;identification&apos;)
    at Object.identification (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:335:33)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › POST /api/entreprises" name="ne devrait pas permettre à un utilisateur entreprise de créer des entreprises" time="0.351">
      <failure>Error: expected 403 &quot;Forbidden&quot;, got 201 &quot;Created&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:365:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › POST /api/entreprises" name="devrait valider les champs obligatoires pour la création" time="0.282">
      <failure>Error: expected 400 &quot;Bad Request&quot;, got 500 &quot;Internal Server Error&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:382:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › DELETE /api/entreprises/:id" name="devrait supprimer une entreprise pour l&apos;administrateur" time="0.362">
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › DELETE /api/entreprises/:id" name="ne devrait pas permettre à un utilisateur entreprise de supprimer des entreprises" time="0.339">
      <failure>Error: expected 403 &quot;Forbidden&quot;, got 200 &quot;OK&quot;
    at Object.expect (D:\api-gestion-main\New folder\server\tests\integration\entreprises\entreprise.crud.test.js:409:10)
    at Promise.finally.completed (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1557:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1497:10)
    at _callCircusTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1007:40)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at _runTest (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:947:3)
    at D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:849:7
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:862:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at _runTestsForDescribeBlock (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:857:11)
    at run (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:761:3)
    at runAndTransformResultsToJestFormat (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\jestAdapterInit.js:1918:21)
    at jestAdapter (D:\api-gestion-main\New folder\server\node_modules\jest-circus\build\runner.js:101:19)
    at runTestInternal (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:275:16)
    at runTest (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:343:7)
    at Object.worker (D:\api-gestion-main\New folder\server\node_modules\jest-runner\build\testWorker.js:497:12)
----
    at Test._assertStatus (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:309:14)
    at D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:365:13
    at Test._assertFunction (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:342:13)
    at Test.assert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:195:23)
    at localAssert (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:138:14)
    at Server.&lt;anonymous&gt; (D:\api-gestion-main\New folder\server\node_modules\supertest\lib\test.js:152:11)
    at Object.onceWrapper (node:events:632:28)
    at Server.emit (node:events:518:28)
    at emitCloseNT (node:net:2418:8)
    at processTicksAndRejections (node:internal/process/task_queues:89:21)</failure>
    </testcase>
    <testcase classname="Tests d&apos;Intégration CRUD Entreprise › DELETE /api/entreprises/:id" name="devrait retourner 404 pour une entreprise inexistante lors de la suppression" time="0.368">
    </testcase>
  </testsuite>
</testsuites>